packages/headless/src/features/generated-answer/generated-answer-slice.ts

```diff
@@ -0,0 +1,42 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {RETRYABLE_STREAM_ERROR_CODE} from '../../api/generated-answer/generated-answer-client';
+import './generated-answer-actions';
+import {
+  resetAnswer,
+  setIsLoading,
+  updateCitations,
+  updateError,
+  updateMessage,
+} from './generated-answer-actions';
+import {getGeneratedAnswerInitialState} from './generated-answer-state';
+
+export const generatedAnswerReducer = createReducer(
+  getGeneratedAnswerInitialState(),
+  (builder) =>
+    builder
+      .addCase(updateMessage, (state, {payload}) => {
```

Comment:
Is it possible to have the `updateMessage` or `updateCitations` actions called after `updateError`? If yes, we may want to clear `state.error`.

Reply:
We could do it for safety. But to answer your question, no currently when the onError is called we also terminate the stream. But there's nothing forcing those two actions to be coupled it's just what the flow is now.

Reply:
I reset the error state in those reducers 👍